src/components/Card/LaptopCard.tsx

```diff
@@ -1,75 +1,107 @@
+import { IAccessory } from '@/interfaces/interfaces';
 import { BasketIcon } from '@/assets/icons/BasketIcon';
-import { rateEmptyImg, rateImg } from '@/assets/constants';
-import { ICard } from '@/interfaces/interfaces';
-import { FC } from 'react';
+import classNames from 'classnames';
 
-interface ILaptopCardProps {
-  product: ICard;
+interface ILaptopCard {
+  product: IAccessory;
+  classname: string;
 }
 
-export const LaptopCard: FC<ILaptopCardProps> = ({ product }) => {
+export const LaptopCard: FC<ILaptopCard> = ({ product, classname }) => {
   const { isLiked, handleClickBuy, handleClickLike } = useProductCardHandlers();
 
   const productRating = product.rate ?? 0;
+  const encodedTitle = encodeURIComponent(product?.title || '');
 
   return (
-    <div>
-      {laptopData.map((item) => (
-        <Card className={styles.cardConatiner} key={item.id}>
-          <div className={styles.cardContainerTop}>
+    <>
+      <Link
+        className={`${styles.cardConatiner} ${classname} `}
+        key={product?.id}
+        to={`/smartphone/${encodedTitle}`}
+        tabIndex={0}
+      >
+        <div
+          className={`${styles.cardContainerTop} ${
+            Array.isArray(product?.hasAnotherColor)
+              ? styles.alignCenter
+              : styles.alignStart
+          }`}
+        >
+          <div className={styles.productImage}>
+            <img src={product?.img} alt="Product image" />
+          </div>
+
+          <div
+            className={`${styles.cardConatinerLike} ${
+              !Array.isArray(product?.hasAnotherColor) ? styles.marginTop : ''
+            }`}
+          >
             <img
-              src={item.img}
-              className={styles.productImage}
-              alt="Product image"
+              onClick={handleClickLike}
+              src={isLiked ? product?.likeIcon : product?.likeIconClick}
+              alt="Like"
+              className={styles.likeIcon}
             />
-            <div
-              className={`${styles.cardConatinerLike} ${styles.cardConatinerLikeLaptop}`}
-            >
-              <img
-                onClick={handleClickLike}
-                src={isLiked ? item.likeIcon : item.likeIconClick}
-                alt="Like"
-                className={styles.likeIcon}
-              />
+            {product?.colorPalette ? (
               <img
-                src={item.colorPalette}
-                className={`${styles.colorPalette} ${styles.colorPaletteLaptop}`}
+                src={product?.colorPalette}
+                className={styles.colorPalette}
                 alt="Palette image"
               />
-            </div>
+            ) : (
+              Array.isArray(product?.hasAnotherColor) &&
+              product?.hasAnotherColor.length > 0 && (
+                <div className={styles['accessories-colors']}>
+                  {product?.hasAnotherColor.map((color: string) => (
```

Comment:
Variables with 'has' and 'is' prefixes should only hold boolean values.
I suggest modifying the IAccessory property.
Remove hasAnotherColor and add anotherColors, which will always be an array, either empty or containing values.